### ISSUE TITLE
coveragepy 7.0 breaks virtualenv CI

### ISSUE THREAD

**Issue by gaborbernat** [coveragepy/coveragepy#1511]
Hey, not sure why and what, but 7 of coverage broke virtualenv: https://github.com/pypa/virtualenv/actions/runs/3747468248/jobs/6363725929

This is the last good version https://github.com/pypa/virtualenv/actions/runs/3723905258/jobs/6315709040

A lot of these showed up too, are we loosing coverage annotation (aka what tested what?)
```bash
Skipping duplicate data .tox/.coverage.py311.fv-az186-962.4986.507187
Skipping duplicate data .tox/.coverage.py311.fv-az186-962.4872.961381
Skipping duplicate data .tox/.coverage.py311.fv-az186-962.5108.664082
```

And ultimately, the CI fails with:
```bash
py311: commands[3]> python -m coverage report --skip-covered --show-missing
No source for code: '/home/runner/work/virtualenv/virtualenv/.tox/py311/lib/python3.11/site-packages/virtualenv/seed/wheels/embed/pip-22.3.1-py3-none-any.whl/pip/__init__.py'.
```

The config should remap this file to src https://github.com/pypa/virtualenv/blob/main/.coveragerc#L9-L24, but does not
the config seems correct to me 🤔:

```ini
omit =
    # site.py is ran before the coverage can be enabled, no way to measure coverage on this
    src/virtualenv/create/via_global_ref/builtin/python2/site.py
    src/virtualenv/create/via_global_ref/_virtualenv.py
    src/virtualenv/activation/python/activate_this.py
    src/virtualenv/seed/wheels/embed/pip-*.whl/*
[coverage:paths]
source =
    src
    .tox/*/lib/python*/site-packages
    .tox/pypy*/site-packages
    .tox\*\Lib\site-packages
    .tox\py\site-packages
    **/src
    **\src
```

**Comment by nedbat** [comment]
I added some debugging of the `coverage combine` path mapping in this run: https://github.com/nedbat/virtualenv/actions/runs/3753744728/jobs/6377291474.   Can you see a line there that is getting the wrong answer?

**Comment by gaborbernat** [comment]
Doesn't help much:

```
No rules match, path '/home/runner/work/virtualenv/virtualenv/.tox/py311/lib/python3.11/site-packages/virtualenv/seed/wheels/embed/pip-22.3.1-py3-none-any.whl/pip/_internal/resolution/resolvelib/reporter.py' is unchanged
Matched path '/home/runner/work/virtualenv/virtualenv/src/virtualenv/create/debug.py' to rule '**/src' -> 'src/', producing '/home/runner/work/virtualenv/virtualenv/src/virtualenv/create/debug.py'
py311: commands[3]> python -m coverage report --skip-covered --show-missing
No source for code: '/home/runner/work/virtualenv/virtualenv/.tox/py311/lib/python3.11/site-packages/virtualenv/seed/wheels/embed/pip-22.3.1-py3-none-any.whl/pip/__init__.py'.
```

as it doesn't say why https://github.com/pypa/virtualenv/blob/main/.coveragerc#L19 does not match 🤔

**Comment by nedbat** [comment]
I found the problem: the path mapping in 7.0 insists that the file exist.  It knew how to handle "paths" of files in .zip files (and .pex and .egg), but it wasn't understanding .whl files.  

The fix is coming and looks good: https://github.com/nedbat/virtualenv/actions/runs/3768230460/jobs/6406493448

But you are supporting Python 3.6, which coverage hasn't supported since 6.3.  How have you been able to run your tests up to now?

**Comment by nedbat** [comment]
This is fixed in commit da1b282d.

**Comment by nedbat** [comment]
This is now released as part of [coverage 7.0.1](https://pypi.org/project/coverage/7.0.1).
